Meta/CMake/wasm_spec_tests.cmake

```diff
@@ -18,15 +17,21 @@ if(INCLUDE_WASM_SPEC_TESTS)
         set(SKIP_PRETTIER true)
     endif()
 
-    if(EXISTS ${WASM_SPEC_TEST_GZ_PATH} AND NOT EXISTS ${WASM_SPEC_TEST_PATH})
+    find_program(WAT2WASM wat2wasm REQUIRED)
+    find_program(PRETTIER prettier OPTIONAL)
+    if (NOT SKIP_PRETTIER AND NOT PRETTIER_FOUND)
+       message(FATAL_ERROR "Prettier required to format Wasm spec tests! Install prettier or set WASM_SPEC_TEST_SKIP_FORMATTING to ON")
+    endif()
+
+    if(EXISTS ${WASM_SPEC_TEST_GZ_PATH} AND NOT EXISTS ${WASM_SPEC_TEST_PATH}/const_0.wasm)
         message(STATUS "Extracting the WebAssembly testsuite from ${WASM_SPEC_TEST_GZ_PATH}...")
         file(MAKE_DIRECTORY ${WASM_SPEC_TEST_PATH})
         if (CMAKE_VERSION VERSION_LESS 3.18.0)
             execute_process(COMMAND "${TAR_TOOL}" -xzf ${WASM_SPEC_TEST_GZ_PATH})
         else()
             file(ARCHIVE_EXTRACT INPUT ${WASM_SPEC_TEST_GZ_PATH} )
         endif()
-        file(GLOB WASM_TESTS "${CMAKE_BINARY_DIR}/testsuite-main/*.wast")
+        file(GLOB WASM_TESTS "${CMAKE_CURRENT_BINARY_DIR}/testsuite-main/*.wast")
```

Comment:
@alimpfard I think if we only change *this* line, it will fix the wasm spec tests in the test262 runner? But I like the rest of this cleanup too :)